### DISCUSSION TITLE
Mcu unable to connect

### DISCUSSION THREAD

**Discussion by seenit783** [bassamanator/everything-sovol-sv06#42] [General]
I foolishly thought I could handle the issues that would arise and I updated klipper to .118 and that's when I couldn't upload  .stl filed, but I resolved that then it let me upload but wouldn't open. Fixed that but in the middle of that I can't get to klipper scren but klipper gives me this![Screenshot_20240311_184756_Facebook.jpg](https://github.com/bassamanator/everything-sovol-sv06/assets/158094770/db4d669e-d06b-48e4-ad19-fe104113ab93)

I have gone through the setup process trying to over write eisting files in hopes of correcting issue. I have tried comparing paths and setting from backups but to no avail still dead in water.

**Comment by bassamanator** [comment]
What devices are in play here (all of them)?

**Reply by seenit783** [reply]
Btt pad 7, sv06+ input shaper

**Comment by bassamanator** [comment]
SSH into the device (pad7). Type the following and paste the output in a [code block](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#fenced-code-blocks).

```sh
cd ~
tree -C -L 1 -F --dirsfirst
```

If you don't have the tree command installed, do a `sudo apt update && sudo apt install tree`.

**Reply by seenit783** [reply]
Ok, I'll do that when I get home I don't recall seeing the install tree part when I was troubleshooting.

**Reply by bassamanator** [reply]
`tree` is a just a basic linux command. I'm trying to determine what the directory structure of this device is so that I can guide you to where to make the edits, copy files, etc.

Take your time.

**Reply by seenit783** [reply]
Ran cmd's installed tree

**Reply by seenit783** [reply]
I'm trying to follow and understand, not sure what you meant to do after whe. You said paste the output in code block 
![20240312_021815](https://github.com/bassamanator/everything-sovol-sv06/assets/158094770/2fdfbffa-45b1-44c3-a2f4-541b505f0e54)

**Reply by seenit783** [reply]
I tried those cmds and it output a really long tree  where I can't scroll up to beginning.. how do I get whole tree so you can view?

**Comment by seenit783** [comment]
[treeC.txt](https://github.com/bassamanator/everything-sovol-sv06/files/14569215/treeC.txt)
[treedirsfirst.txt](https://github.com/bassamanator/everything-sovol-sv06/files/14569277/treedirsfirst.txt)
[treef.txt](https://github.com/bassamanator/everything-sovol-sv06/files/14569279/treef.txt)
[treel.txt](https://github.com/bassamanator/everything-sovol-sv06/files/14569280/treel.txt)

**Comment by bassamanator** [comment]
Be careful with the commands and type them into the terminal as is.

What is the output from?
```sh
ls /dev/serial/by-id/*
```
Alternate method if the above doesn't yield any results.
```sh
ls /dev/serial/by-path/*
```

**Reply by seenit783** [reply]
works on both commands for me.

**Reply by seenit783** [reply]
[31224114.txt](https://github.com/bassamanator/everything-sovol-sv06/files/14577098/31224114.txt)

**Comment by bassamanator** [comment]
Please post future output here, not in txt files.

Follow instructions [from here](https://github.com/bassamanator/Sovol-SV06-firmware?tab=readme-ov-file#initial-steps) and adjust your `mcu` path accordingly.

**Reply by seenit783** [reply]
sorry, i'm still a newbie apparently. i just doing what i know to do to get the information to you. Thank you.

**Reply by seenit783** [reply]
updated printer. cfg and mainsail. cfg files and restarted. same error.

**Comment by bassamanator** [comment]
> sorry, i'm still a newbie apparently. i just doing what i know to do to get the information to you. Thank you.

No worries.

> updated printer. cfg and mainsail. cfg files and restarted. same error.

What did you change in mainsail.cfg?

Attach `printer.cfg`, `moonraker.log` and `klippy.log` here, find them through your dashboard.

**Reply by seenit783** [reply]
I copied path and placed it where the other path was that I had been using before I broke the code. I'll get you logs early in the morning about 4am cst

**Reply by bassamanator** [reply]
Okay, but there were no changes required to mainsail.cfg, so undo whatever you've done there. `mcu` path is only added to `printer.cfg`.

No rush.

**Reply by seenit783** [reply]
Ah, I'm sorry. I will undo that, I was told in past to do that and or I mis interpreted them.

**Reply by bassamanator** [reply]
Don't stress my friend, this is a fixable issue. Get back to me when you have time.

**Reply by seenit783** [reply]
[klippy.log](https://github.com/bassamanator/everything-sovol-sv06/files/14586183/klippy.log)
[moonraker.log](https://github.com/bassamanator/everything-sovol-sv06/files/14586184/moonraker.log)

**Reply by seenit783** [reply]
it wouldn't let me upload the .cfg format so i had to change file name
[printer.log](https://github.com/bassamanator/everything-sovol-sv06/files/14586238/printer.log)

**Comment by bassamanator** [comment]
I believe this [video](https://youtu.be/cnthbR3tS9M?si=ZtYgP07hsh9jhBIA) will solve your issue, but you're only going to be doing the second part of it, starting at 1:05.

**Reply by seenit783** [reply]
I feel like this guy skipped steps in video. He assumed the person knew software well enough he could just say what was needed.

**Reply by seenit783** [reply]
Also feel because of the screen shot he has that I have a separate issue. He has a mcu protocol issue which I fixed mine says mcu unable to connect.

**Reply by bassamanator** [reply]
What steps did he skip? He literally showed you every step that is required. Also, as I said, you need to do the second part only, because I never said you two have the same issue, only that 
> this [video](https://youtu.be/cnthbR3tS9M?si=ZtYgP07hsh9jhBIA) will solve your issue

And again, I'm coming to this conclusion based on very little data, given that I don't own the device, and have never used it. But this is where my mind is going right now.

**Reply by seenit783** [reply]
So basically I figured out what was wrong... I did nothing different before on flashing printer and everything except this time I walked away from printer while I was flashing it and I guess I didn't allow it enough time to flash each time I tried.

**Reply by seenit783** [reply]
I deeply appreciate your help, it made me feel better knowing that I was on the right track and it ended up me being impatient on flashing printer. I guess my 20 seconds wasn't enough.

**Reply by bassamanator** [reply]
Welcome! Please close this discussion when ready.

**Reply by seenit783** [reply]
Uploading 20240313_162751.mp4…

**Comment by seenit783** [comment]
So long story short, we were doing everything right but 3d printer wanted longer amount of time to flash .bin file than was normally given. Waited about 60 seconds.

**Reply by bassamanator** [reply]
👍🏼 Don't forget to star this repo! Thanks!
